graph/articles/collections.md

```diff
@@ -316,3 +316,148 @@ Content-Type: application/json
+Any changes to one type must be reflected in the other type.
+2. Reject requests that update both collections at the same time.
+A request that adds an item to `barsAsEntities` while replacing the content of `bars` must rejected with a `400`, for example:
+```http
+PATCH /foos/{fooId}
+{
+  "bars": [
+    {
+      "prop1": "some value",
+      "prop2": "another value"
+    }
+  ],
+  "barsAsEntities@delta": [
+    {
+      "prop1": "a key value",
+      "prop2": "some new value"
+    }
+  ]
+}
+```
+```http
+HTTP/1.1 400 Bad Request
+{
+  "error": {
+    "code": "badRequest",
+    "message": "'bars' and 'barsAsEntities' cannot be updated in the same request.",
+}
+```
+
+TODO should this be a 409 conflict instead?
+TODO implement this in WebApi
+
+### 11.2 `$select` overloading
+
+The model can be updated to simply switch the complex type for an entity type:
+```diff
+<EntityType Name="foo">
+  <Key>
+    <PropertyRef Name="id" />
+  </Key>
+  <Property Name="id" Type="Edm.String" Nullable="false" />
+- <Property Name="bars" Type="Collection(self.bar)" />
++ <NavigationProperty Name="bars" Type="Collection(self.bar)" ContainsTarget="true" />
+</EntityType>
+
+- <ComplexType Name="bar">
++ <EntityType Name="bar">
++ <Key>
++   <PropertyRef Name="prop1" />
++ </Key>
+  <Property Name="prop1" Type="Edm.String" />
+  <Property Name="prop2" Type="Edm.String" />
+-</ComplexType>
++</EntityType>
+```
+To maintain backwards compatibility **and** compliance with the OData standard, there are several semantic changes that the workload must address:
+1. Existing clients would have been able to `$select` the `bars` property.
+Now that `bars` is a navigation property, the [OData standard](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#_Toc31361040) specifies that its navigation link be returned when it is `$selected`:
+
+> If the select item is a navigation property, then the corresponding navigation link is represented in the response.
+
+Because the previous behavior for `$select=bars` was to include the collection in the response, and because the standard dictates that the navigation link be included in the response, the new behavior is to include both:
+
+```http
+GET /foos/{fooId}?$select=bars
+```
+```http
+200 OK
+{
+  "id": "{fooId}",
+  "bars": [
+    {
+      "prop1": "some value",
+      "prop2": "another value"
+    },
+    ...
+  ]
+  "bars@odata.navigationLink": "/foos('{fooId}')/bars"
+}
+```
+
+2. The default behavior for structural collections is to include them in the response payload for their containing entity. If this was the behavior of `foo` before, it must be preserved by **auto-expanding** the `bars` property now that it is a navigation property (because the default behavior for navigation properties is to **not** expand them).
+3. Structural collections are updated using `PATCH` requests to replace the entire contents of the collection. The new navigation property must preserve this behavior.
```

Comment:
to be clear, the patch request is to the containing entity (not the collection), and this is required only if the original structural property supported updating.
```suggestion
3. Structural collections can be updated using a `PATCH` request to the containing entity to replace the entire contents of the collection. If the service supported such updates to the structural collection, then updates to the new navigation property must preserve this behavior.
```

Reply:
I'm concerned about adding too many intervening clauses and conditionals. Workload teams need to be able to understand what is expected of them in a straightforward manner. 

I don't think anything in my original statements is factually incorrect. I would love to correct anything that's wrong, but we need to do so in a way that doesn't make this sound "optional".